src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -216,7 +216,7 @@ public String getRemoteFs() {
     }
 
     public void setInstanceCap(int instanceCap) {
-        if (instanceCap <= 0) {
+        if (instanceCap < 0) {
```

Comment:
if you had instanceCap=0 the xml is written with `<instanceCap>2147483647</instanceCap>` so if I'm not mistaken there is nothing to do